src/databricks/labs/ucx/azure/resources.py

```diff
@@ -222,23 +223,26 @@ def __init__(self, host_endpoint: str, service_endpoint: str):
         self.api_client = ApiClient(
             Config(
                 host=host_endpoint,
-                credentials_provider=self._provider_for(service_endpoint),
+                credentials_strategy=self._provider_for(service_endpoint),
             )
         )
         self._token_source = AzureCliTokenSource(host_endpoint)
 
-    def _provider_for(self, endpoint: str):
-        @credentials_provider("azure-cli", ["host"])
-        def _credentials(_: Config):
+    @staticmethod
+    def _provider_for(endpoint: str) -> CredentialsStrategy:
+        @credentials_strategy("azure-cli", ["host"])
+        def _credentials(_: Config) -> CredentialsStrategy:
```

Comment:
Should the return type here be `CredentialsProvider`?

Reply:
Nope, the `Provider` is renamed to `Strategy` in the latest SDK version

Reply:
Are you sure?

I think `CredentialsStrategy` is new in addition to `CredentialsProvider` rather than replacing it, as can be seen [here](https://github.com/databricks/databricks-sdk-py/blob/228cc8f5ef13eab93cd5f5c082c4ca4bca61ab1d/databricks/sdk/credentials_provider.py#L80).

If I look at the functions annotated with `@credentials_strategy()` in the same file they all return `CredentialsProvider`.

Reply:
See [https://github.com/databricks/databricks-sdk-py/blob/228cc8f5ef13eab93cd5f5c082c4ca4bca61ab1d/databricks/sdk/config.py#L111-L113]:

```
                 # Deprecated. Use credentials_strategy instead.
                 credentials_provider: CredentialsStrategy = None,
                 credentials_strategy: CredentialsStrategy = None,
```